pkg/server/cert/cert.go

```diff
@@ -3,14 +3,15 @@ package cert
 import (
 	"crypto"
 	"crypto/x509"
-	"github.com/loft-sh/vcluster/pkg/util/certhelper"
+	"fmt"
 	"io/ioutil"
-	"k8s.io/apimachinery/pkg/util/sets"
 	"net"
-	"os"
+
+	"github.com/loft-sh/vcluster/pkg/util/certhelper"
+	"k8s.io/apimachinery/pkg/util/sets"
 )
 
-func GenServingCerts(caCertFile, caKeyFile, certFile, keyFile, clusterDomain string, SANs []string) (bool, error) {
+func GenServingCerts(caCertFile, caKeyFile string, currentCert, currentKey *[]byte, clusterDomain string, SANs []string) (bool, error) {
```

Comment:
please return new key and certificate instead of replacing them in-place in the parameters